plugins/flatpak/gs-flatpak.c

```diff
@@ -3679,72 +3680,64 @@ app_has_local_source (GsApp *app)
+get_installation_dir_free_space (GsFlatpak *self, GError **error)
 {
 	g_autoptr (GFile) installation_dir = NULL;
 	g_autoptr (GFileInfo) info = NULL;
-	g_autoptr (GError) local_error = NULL;
-	guint64 free_space = 0;
 
 	installation_dir = flatpak_installation_get_path (self->installation);
 
 	info = g_file_query_filesystem_info (installation_dir,
 					     G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
 					     NULL,
-					     &local_error);
-	if (info == NULL)
-		g_warning ("Failed to query filesystem info: %s", local_error->message);
+					     error);
+	if (info == NULL) {
+		g_warning ("Failed to query filesystem info");
+		return 0;
+	}
 
-	free_space = g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_FREE);
-	return free_space;
+	return g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_FREE);
 }
 
 static gboolean
 gs_flatpak_has_space_to_install (GsFlatpak *self, GsApp *app, GsAppList *list)
 {
+	g_autoptr(GError) error = NULL;
 	guint64 free_space = 0;
 	guint64 space_required = 0;
 
 	space_required = gs_app_get_size_download (app);
 	if (space_required == GS_APP_SIZE_UNKNOWABLE)
 		g_warning ("Failed to query download size: %s", gs_app_get_unique_id (app));
 
-	space_required = space_required + (500 * 1024 * 1024); /* min-free-space-size = 500MB for Endless */
-	free_space = get_installation_dir_free_space (self);
-
-	if (free_space < space_required)
-		return FALSE;
+	space_required = space_required + EOS_MIN_FREE_SPACE_NEEDED_BYTES;
+	free_space = get_installation_dir_free_space (self, &error);
+	if (error != NULL)
+		return TRUE;
 
-	return TRUE;
+	return free_space >= space_required;
 }
 
 static gboolean
-gs_flatpak_has_space_to_update (GsFlatpak *self, GsApp *app, GsAppList *list)
+gs_flatpak_has_space_to_update (GsFlatpak *self, GsApp *app, GsAppList *list, gboolean is_auto_update)
 {
+	g_autoptr(GError) error = NULL;
 	guint64 free_space = 0;
 	guint64 space_required = 0;
 
-	/* If an app has an updatable runtime, the runtime is not updated
-	 * as a part of app update.
-	 *
-	 * Having said that, runtime updates and their related refs (see T22716)
-	 * are done together and below heuristics might not be good for if GsApp
-	 * app is a runtime.
-	 */
-	if (gs_utils_app_is_auto_updating (app)) {
+	if (is_auto_update && gs_app_get_kind (app) != AS_APP_KIND_RUNTIME) {
 		for (guint i = 0; i < gs_app_list_length (list); i++) {
 			GsApp *app_temp = gs_app_list_index (list, i);
 			space_required += gs_app_get_size_installed (app_temp);
 		}
 	}
 
-	space_required = (space_required * 2) + (500 * 1024 * 1024); /* min-free-space-size = 500MB for Endless */
-	free_space = get_installation_dir_free_space (self);
-
-	if (free_space < space_required)
-		return FALSE;
+	space_required = (space_required * 2) + EOS_MIN_FREE_SPACE_NEEDED_BYTES;
+	free_space = get_installation_dir_free_space (self, &error);
+	if (error != NULL)
+		return is_auto_update ? FALSE : TRUE;
```

Comment:
Add a comment here explaining why we differ when an error occurs (otherwise in 2 weeks we'll have to waste brain power just understanding why we did it).